packages/compiler/src/server/diagnostics.ts

```diff
@@ -127,11 +160,39 @@ interface VSLocation {
   readonly range: Range;
 }
 
-function getVSLocation(
+async function getVSLocation(
   fileService: FileService,
   location: SourceLocation | undefined,
   currentDocument: TextDocument,
-): VSLocation | undefined {
+  configFilePath: string | undefined = undefined,
+  emitterName: string | undefined = undefined,
+): Promise<VSLocation | typeof NoTarget | undefined> {
+  if (configFilePath && emitterName && emitterName.length > 0 && location === undefined) {
+    const [yamlScript] = parseYaml(await NodeSystemHost.readFile(configFilePath));
```

Comment:
why do we need to load the yaml here? I think we did the same thing for the import error in tspconfig.yaml but didn't need this change.